Tests added fc893c8 probably need "non-bundled cfitsio" marker

I see that the tests added in fc893c8 expect cfitsio compiled without FMA instructions, and they fail for me against system cfitsio on amd64 Linux:

```pytb
============================================================== FAILURES ===============================================================
_________________________________________ test_image_compression_write_read_comp_to_osx_arm64 _________________________________________

    def test_image_compression_write_read_comp_to_osx_arm64():
        pth = os.path.join(
            os.path.dirname(__file__),
            "..",
            "test_images",
            "test_rice_dither2_seed42.fits",
        )
        seed = 42
        rng = np.random.RandomState(seed=seed)
        data = rng.normal(size=(100, 100))
    
        with tempfile.TemporaryDirectory() as tmpdir:
            tpth = os.path.join(tmpdir, f"test_rice_dither2_seed{seed}.fits.fz")
            write(
                tpth,
                data,
                compress="RICE",
                qmethod="SUBTRACTIVE_DITHER_2",
                dither_seed=seed,
            )
            tdata = read(tpth)
        data = read(pth)
    
>       np.testing.assert_array_equal(data, tdata)
E       AssertionError: 
E       Arrays are not equal
E       
E       Mismatched elements: 10000 / 10000 (100%)
E       First 5 mismatches are at indices:
E        [0, 0]: 0.4631004333496094 (ACTUAL), 0.46310049088085103 (DESIRED)
E        [0, 1]: -0.0413132905960083 (ACTUAL), -0.04131327904654449 (DESIRED)
E        [0, 2]: 0.7431849241256714 (ACTUAL), 0.7431849415876269 (DESIRED)
E        [0, 3]: 1.51054048538208 (ACTUAL), 1.5105404973206271 (DESIRED)
E        [0, 4]: -0.13693153858184814 (ACTUAL), -0.13693156157350245 (DESIRED)
E       Max absolute difference among violations: 1.78749362e-07
E       Max relative difference among violations: 0.00016445
E        ACTUAL: array([[ 0.4631  , -0.041313,  0.743185, ...,  0.352413, -0.024024,
E               -0.14413 ],
E              [-1.363684, -0.528113, -0.387114, ...,  0.147066,  0.108465,...
E        DESIRED: array([[ 0.4631  , -0.041313,  0.743185, ...,  0.352413, -0.024024,
E               -0.144129],
E              [-1.363684, -0.528113, -0.387114, ...,  0.147066,  0.108465,...

data       = array([[ 0.46310043, -0.04131329,  0.74318492, ...,  0.3524127 ,
        -0.02402365, -0.14412951],
       [-1.36368418, -0.52811259, -0.38711417, ...,  0.14706576,
         0.10846543, -1.09521443],
       [ 0.45728242,  0.52508879,  0.98870659, ...,  0.34022939,
         0.73443723,  0.52627134],
       ...,
       [-0.28811377,  0.38840121,  0.6416744 , ...,  0.26547617,
        -0.17389315, -1.22321039],
       [ 0.97257656,  0.49103165,  2.37570399, ..., -0.33387434,
         1.67651981,  0.95810032],
       [ 1.03601933,  0.2497189 , -1.11601806, ..., -0.81380963,
         0.55565405,  0.6619699 ]], shape=(100, 100))
pth        = '/tmp/portage/dev-python/fitsio-1.4.0/work/fitsio-1.4.0-python3_12/install/usr/lib/python3.12/site-packages/fitsio/tests/../test_images/test_rice_dither2_seed42.fits'
rng        = RandomState(MT19937) at 0x7F772983CF40
seed       = 42
tdata      = array([[ 0.46310049, -0.04131328,  0.74318494, ...,  0.35241276,
        -0.02402362, -0.14412949],
       [-1.36368418, -0.52811261, -0.38711419, ...,  0.14706578,
         0.10846546, -1.09521445],
       [ 0.45728234,  0.52508862,  0.98870641, ...,  0.34022931,
         0.73443705,  0.52627128],
       ...,
       [-0.28811375,  0.3884012 ,  0.64167438, ...,  0.26547617,
        -0.17389316, -1.22321039],
       [ 0.97257651,  0.4910316 ,  2.37570394, ..., -0.33387442,
         1.67651973,  0.95810025],
       [ 1.03601929,  0.24971887, -1.11601802, ..., -0.81380961,
         0.55565409,  0.66196988]], shape=(100, 100))
tmpdir     = '/tmp/portage/dev-python/fitsio-1.4.0/temp/tmpmmtizqyp'
tpth       = '/tmp/portage/dev-python/fitsio-1.4.0/temp/tmpmmtizqyp/test_rice_dither2_seed42.fits.fz'

fitsio/tests/test_image_compression.py:868: AssertionError
_____________________________________________ test_image_compression_read_from_osx_arm64 ______________________________________________

    def test_image_compression_read_from_osx_arm64():
        pth = os.path.join(
            os.path.dirname(__file__),
            "..",
            "test_images",
            "test_rice_dither2_seed42.fits",
        )
        cpth = os.path.join(
            os.path.dirname(__file__),
            "..",
            "test_images",
            "test_rice_dither2_seed42.fits.fz",
        )
        cdata = read(cpth)
        data = read(pth)
    
>       np.testing.assert_array_equal(data, cdata)
E       AssertionError: 
E       Arrays are not equal
E       
E       Mismatched elements: 10000 / 10000 (100%)
E       First 5 mismatches are at indices:
E        [0, 0]: 0.4631004333496094 (ACTUAL), 0.46310049088085103 (DESIRED)
E        [0, 1]: -0.0413132905960083 (ACTUAL), -0.04131327904654449 (DESIRED)
E        [0, 2]: 0.7431849241256714 (ACTUAL), 0.7431849415876269 (DESIRED)
E        [0, 3]: 1.51054048538208 (ACTUAL), 1.5105404973206271 (DESIRED)
E        [0, 4]: -0.13693153858184814 (ACTUAL), -0.13693156157350245 (DESIRED)
E       Max absolute difference among violations: 5.95946681e-08
E       Max relative difference among violations: 0.00011772
E        ACTUAL: array([[ 0.4631  , -0.041313,  0.743185, ...,  0.352413, -0.024024,
E               -0.14413 ],
E              [-1.363684, -0.528113, -0.387114, ...,  0.147066,  0.108465,...
E        DESIRED: array([[ 0.4631  , -0.041313,  0.743185, ...,  0.352413, -0.024024,
E               -0.144129],
E              [-1.363684, -0.528113, -0.387114, ...,  0.147066,  0.108465,...

cdata      = array([[ 0.46310049, -0.04131328,  0.74318494, ...,  0.35241276,
        -0.02402362, -0.14412949],
       [-1.36368418, -0.52811261, -0.38711419, ...,  0.14706578,
         0.10846546, -1.09521445],
       [ 0.45728246,  0.52508874,  0.98870653, ...,  0.34022943,
         0.73443717,  0.5262714 ],
       ...,
       [-0.28811375,  0.3884012 ,  0.64167438, ...,  0.26547617,
        -0.17389316, -1.22321039],
       [ 0.97257657,  0.49103166,  2.375704  , ..., -0.33387436,
         1.67651979,  0.95810031],
       [ 1.03601929,  0.24971887, -1.11601802, ..., -0.81380961,
         0.55565409,  0.66196988]], shape=(100, 100))
cpth       = '/tmp/portage/dev-python/fitsio-1.4.0/work/fitsio-1.4.0-python3_12/install/usr/lib/python3.12/site-packages/fitsio/tests/../test_images/test_rice_dither2_seed42.fits.fz'
data       = array([[ 0.46310043, -0.04131329,  0.74318492, ...,  0.3524127 ,
        -0.02402365, -0.14412951],
       [-1.36368418, -0.52811259, -0.38711417, ...,  0.14706576,
         0.10846543, -1.09521443],
       [ 0.45728242,  0.52508879,  0.98870659, ...,  0.34022939,
         0.73443723,  0.52627134],
       ...,
       [-0.28811377,  0.38840121,  0.6416744 , ...,  0.26547617,
        -0.17389315, -1.22321039],
       [ 0.97257656,  0.49103165,  2.37570399, ..., -0.33387434,
         1.67651981,  0.95810032],
       [ 1.03601933,  0.2497189 , -1.11601806, ..., -0.81380963,
         0.55565405,  0.6619699 ]], shape=(100, 100))
pth        = '/tmp/portage/dev-python/fitsio-1.4.0/work/fitsio-1.4.0-python3_12/install/usr/lib/python3.12/site-packages/fitsio/tests/../test_images/test_rice_dither2_seed42.fits'

fitsio/tests/test_image_compression.py:887: AssertionError
```

I see that some tests are set to XFAIL when using the system library. Could we apply a similar treatment here?

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -11,6 +11,8 @@ Changes
 Bug Fixes
 
 - Fixed (rare) bug where `FITS.hdu_list` and `FITS.hdu_map` can go out of sync.
+- Marked tests for reproducible lossy float compression as xfail for external
+  cfitsio libraries.
 
 ## 1.4.0
 

--- a/fitsio/tests/test_image_compression.py
+++ b/fitsio/tests/test_image_compression.py
@@ -842,6 +842,13 @@ def test_image_compression_read_chunks():
                 assert np.all(read_data == data[start:end])
 
 
+@pytest.mark.xfail(
+    condition=not cfitsio_is_bundled(),
+    reason=(
+        "system cfitsio must be compiled without FMA instructions to "
+        "enable reproducible lossy float compression"
+    ),
+)
 def test_image_compression_write_read_comp_to_osx_arm64():
     pth = os.path.join(
         os.path.dirname(__file__),
@@ -868,6 +875,13 @@ def test_image_compression_write_read_comp_to_osx_arm64():
     np.testing.assert_array_equal(data, tdata)
 
 
+@pytest.mark.xfail(
+    condition=not cfitsio_is_bundled(),
+    reason=(
+        "system cfitsio must be compiled without FMA instructions to "
+        "enable reproducible lossy float compression"
+    ),
+)
 def test_image_compression_read_from_osx_arm64():
     pth = os.path.join(
         os.path.dirname(__file__),

